src/sortStudents.ts

```diff
@@ -1,16 +1,91 @@
 
 export interface Student {
   // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
   // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
-
-export function sortStudents(students, sortBy, order) {
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
   // write your function
+
+  const studentsCopy: Student[] = [...students];
+
+  if (order === 'asc') {
+    switch (sortBy) {
+      case SortType.Name:
+        return studentsCopy.sort(
+          (a: Student, b: Student) => a.name.localeCompare(b.name),
+        );
+
+      case SortType.Surname:
+        return studentsCopy.sort(
+          (a: Student, b: Student) => a.surname.localeCompare(b.surname),
```

Comment:
Use sortBy to get a necessary value 
```suggestion
          (a: Student, b: Student) => a[sortBy].localeCompare(b.[sortBy]),
```